### DISCUSSION TITLE
Bigtreetech SKR Pro 1.1/1.2: undefined reference to `board_init'

### DISCUSSION THREAD

**Discussion by acinortele** [grblHAL/STM32F4xx#191] [General]
**Issue: Compiling grblHAL for Bigtreetech SKR Pro 1.1/1.2 Board**

I'm trying to compile grblHAL for the Bigtreetech SKR Pro 1.1 or 1.2 board using STM32Cube with the following parameters:

| Symbol                  | Value    |
|-------------------------|----------|
| _USE_IOCTL              | 1        |
| _USE_WRITE              | 1        |
| _VOLUMES                | 1        |
| BOARD_BTT_SKR_PRO_1_2   | 1        |
| ESTOP_ENABLE            | 0        |
| HAS_BOOTLOADER          |          |
| HSE_VALUE               | 8000000  |
| OVERRIDE_MY_MACHINE     |          |
| PROBE_ENABLE            | 0        |
| STM32F407xx             |          |
| TRINAMIC_ENABLE         | 2130     |
| TRINAMIC_R_SENSE        | 110      |
| USB_SERIAL_CDC          |          |
| USE_HAL_DRIVER          |          |
| Y_AUTO_SQUARE           | 1        |
| Y_GANGED                | 1        |

I am encountering the following error:

```
driver.c:(.text.driver_init+0x446): undefined reference to `board_init`
```

It seems that the `board_init` function has not been defined for these boards. I noticed that some boards have a "board.c" file in addition to the "board.map" file, but the SKR Pro 1.1 and 1.2 do not have one.

For example, the board defined in `longboard32_map.h` has the following code in `longboard32.c`:

```c
#include "driver.h"

#if defined(BOARD_LONGBOARD32)

extern void tmc_spi_init (void);

void board_init (void)
{
#if TRINAMIC_ENABLE
    tmc_spi_init();
#endif
}
```

Could I use a similar solution for the SKR Pro 1.1/1.2? Would the code be the same?

**Comment by terjeio** [comment]
Fix commited, the code is basically the same for as for the Longboard but since the Longboard code will be expanded I added it as shared code to _tmc_spi.c_ instead.
